File: algo_3_two_points/L11-m.cpp
//
// Created by LiHeng on 2020/4/21.
//
//11 盛最多水的容器

#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    int maxArea(vector<int>& height) {
        int res = 0;
        int left = 0, right = height.size()-1;
        while(left<right){
            res = max(res , min(height[left],height[right])*(right-left));
            if(height[left]<height[right])
                ++left;
            else
                --right;
        }
        return res;
    }
};


int main(){
    int a[] = {1,8,6,2,5,4,8,3,7};
    vector<int> v1(a,a+9);
    cout << Solution().maxArea(v1)<<endl;
}

File: python-algorithm/algo_03_two_points/L11-m.py
#--coding:utf-8--

# 这道题的证明...
class Solution:
    def maxArea(self, height: List[int]) -> int:
        lt, rt = 0, len(height)-1
        res = 0
        while lt < rt:
            res = max(res, min(height[lt], height[rt])*(rt-lt))
            if height[lt]<height[rt]:
                lt += 1
            else :
                rt -= 1
        return res